### PR TITLE
SSCS-8967:Handle error for unreadable and encrypted PDF for Action further evidence

### PR DESCRIPTION
**Pull request by divisathyan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-8967

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceMidEventHandler.java [hunk 1]
```diff
@@ -29,32 +42,82 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
+            List<String> unreadablePdfLinks = new ArrayList<>();
+            List<String> encryptedPdfLinks = new ArrayList<>();
             for (ScannedDocument scannedDocument : sscsCaseData.getScannedDocuments()) {
                 if (scannedDocument != null && scannedDocument.getValue() != null) {
-                    checkWarningsAndErrors(sscsCaseData, scannedDocument, sscsCaseData.getCcdCaseId(), ignoreWarnings, preSubmitCallbackResponse);
+                    if (scannedDocument.getValue().getUrl() != null
+                        && scannedDocument.getValue().getUrl().getDocumentUrl() != null) {
+                        try {
+                            footerService.isReadablePdf(scannedDocument.getValue().getUrl().getDocumentUrl());
+                        } catch (PdfPasswordException e) {
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceMidEventHandler.java:75]: This is API is not part of this PR, I know, but this is assuming its going to return a boolean you don't check, and it's throwing two different Exceptions for a password and unreadable pdf. You can't do anything with an API like this apart from write ugly code around it.
It would be better if it returned an enum for PDF_OK, UNREADABLE or HAS_PASSWORD.

Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceMidEventHandler.java:75]: I agree but it is also used in UploadDocumentFurtherEvidenceAboutToSubmitHandler and they rely on catching exception too. If I change the API, we will have to test the other feature as well. The boolean return is not used by the methods.

Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceMidEventHandler.java]: Refactored the code

**Review by danhamilton — APPROVED** [review]
LGTM!

**Comment by divisathyan** [comment]
Code refactored in #1986
